### DISCUSSION TITLE
Auth: User unexpectedly logs out when linking new OAuth provider fails

### DISCUSSION THREAD

**Discussion by rizkianakbar** [supabase/supabase#28688] [Questions]
When attempting to link a new OAuth identity (e.g., Discord) to an existing authenticated user, a failed linking attempt causes an unintended logout. This disrupts the user experience and loses the original authentication state.

### Reproduction Steps
1. User authenticates with Twitter via Supabase Auth.
2. Navigate to account page (e.g., `http://localhost:3000/account`).
3. Attempt to link a Discord account using `supabase.auth.linkIdentity()`.
4. Simulate a failure (e.g., try linking a Discord account already associated with another user).
5. Observe that the user is logged out of their Twitter authentication.

### Expected vs Actual Behavior
- **Expected**: On failed linking, user remains logged in with original (Twitter) auth. An error message is displayed.
- **Actual**: User is logged out. The auth cookie changes from `sb-[project-ref]-auth-token` to `sb-[project-ref]-auth-token-code-verifier`.

### Code Sample
```typescript
const connect = async (provider: Provider) => {
  try {
    const { error } = await supabase.auth.linkIdentity({
      provider,
      options: {
        redirectTo: `${APP_URL}/account`,
      },
    })
    if (error) throw error;
  } catch (error) {
    console.error(error);
    // User is unexpectedly logged out here
  }
}
```

### Error URL
On failed linking:
```
http://localhost:3000/account?error=server_error&error_code=422&error_description=Identity+is+already+linked+to+another+user
```

### Questions
1. Is this logout behavior intended?
2. How can we maintain the original authentication when identity linking fails?
3. Are there recommended practices for handling failed identity linking?

**Comment by christiangenco** [comment]
I've narrowed this down to [the presence of the `error_description` query parameter](https://github.com/supabase/ssr/issues/74) specifically on client-rendered pages.

**Comment by mbaghou** [comment]
Hello,
Having the same issue. any workaround on this pls ?
Thank you
